metronome/hotstuff/service/src/io/iohk/metronome/hotstuff/service/Status.scala

```diff
@@ -0,0 +1,16 @@
+package io.iohk.metronome.hotstuff.service
+
+import io.iohk.metronome.hotstuff.consensus.ViewNumber
+import io.iohk.metronome.hotstuff.consensus.basic.Agreement
+import io.iohk.metronome.hotstuff.consensus.basic.QuorumCertificate
+
+/** Status has all the fields necessary for nodes to sync with each other.
+  *
+  * This is to facilitate nodes rejoining the network,
+  * or re-syncing their views after some network glitch.
+  */
+case class Status[A <: Agreement](
+    viewNumber: ViewNumber,
+    prepareQC: QuorumCertificate[A],
+    commitQC: QuorumCertificate[A]
```

Comment:
Might be a good idea to add `lockedQC`. Although it is difficult to prove that a node is locked to that QC, it might be helpful to handle dynamic validator sets.

Reply:
We can add anything, but I thought we can do so when we know how we'll use them in the synchronisation process, perhaps when we have a spec for the handing of dynamic validator sets? I thought these two are the minimum that is required: the highest `prepareQC` is going to be the basis for the next round, and the `commitQC` is where we can sync from. Not sure how to use `lockedQC` right now.